homeassistant/components/generic_thermostat/climate.py

```diff
@@ -343,15 +345,10 @@ def hvac_modes(self):
         """List of available operation modes."""
         return self._hvac_list
 
-    @property
-    def preset_mode(self):
-        """Return the current preset mode, e.g., home, away, temp."""
-        return PRESET_AWAY if self._is_away else PRESET_NONE
-
     @property
     def preset_modes(self):
         """Return a list of available preset modes or PRESET_NONE if _away_temp is undefined."""
-        return [PRESET_NONE, PRESET_AWAY] if self._away_temp else PRESET_NONE
+        return SUPPORTED_PRESETS if self._away_temp else PRESET_NONE
```

Comment:
This needs to always be a list. Now that you're reuusing `SUPPORTED_PRESETS`, you should make that constant a list again 🙈 

I would create this list when we process config and set `self._away_temp` and assign to `self._attr_preset_modes` and drop this property. You can then also compare to that inside `set_preset_mode`.